[split from #4211] test_misc_xpu: dynamo var_getattr renamed to getattro_impl (test porting)

### 🐛 Describe the bug

**Affected tests (2):**
- `MiscTests.test_inspect_variable_redirect`
- `MiscTests.test_getattrvariable_as_python_constant`

**Root cause:** Two dynamo refactoring PRs landed between the good and bad commits, and the XPU test port `test/xpu/dynamo/test_misc_xpu.py` was not updated:

1. **PR #186013** (commit `18e1ec10f24`) — global rename `var_getattr → getattro_impl` across all `VariableTracker` subclasses. `InspectVariable.var_getattr` no longer exists, causing `AttributeError: type object 'InspectVariable' has no attribute 'var_getattr'` at test line ~12735.

2. **PR #187196** (commit `6d665f54aa9`) — `GetAttrVariable.as_python_constant` is no longer invoked during attribute access; `CallMethodVariable.as_python_constant` is called instead. The XPU test still patches `GetAttrVariable`, so `po.call_count` stays 0, causing `AssertionError: 0 not greater than 0`.

**Fix in `test/xpu/dynamo/test_misc_xpu.py`:**

For `test_inspect_variable_redirect` (~line 12727):
```python
# Before:
original_var_getattr = InspectVariable.var_getattr
def tracking_var_getattr(self, tx, name): ...
with patch.object(InspectVariable, "var_getattr", tracking_var_getattr):
# After:
original_getattro_impl = InspectVariable.getattro_impl
def tracking_getattro_impl(self, tx, name): ...
with patch.object(InspectVariable, "getattro_impl", tracking_getattro_impl):
```

For `test_getattrvariable_as_python_constant` (~line 13140):
```python
# Before:
from torch._dynamo.variables.misc import GetAttrVariable
with patch.object(GetAttrVariable, "as_python_constant", autospec=True) as po:
# After:
from torch._dynamo.variables.misc import CallMethodVariable as CMV
with patch.object(CMV, "as_python_constant", autospec=True) as po:
```

**Reproducer:**
```bash
cd third_party/torch-xpu-ops/test/xpu/dynamo && \
  python -m pytest -sxv test_misc_xpu.py -k "test_inspect_variable_redirect or test_getattrvariable_as_python_constant"
```

### Versions

- PyTorch version: `2.14.0a0+gita0f2bdc`
- Good commit: `de829759c7e622b806788914ed98a6160b2c5f1c`
- Bad commit: `a0f2bdc50c8590762d0c9ff73bd1801898707001`
- Upstream fix commit: `e356f51370f` (PR #188591, 2026-07-02)
- Hardware: Intel Data Center GPU Max 1100

--- a/test/xpu/dynamo/test_misc_xpu.py
+++ b/test/xpu/dynamo/test_misc_xpu.py
@@ -12722,13 +12722,13 @@ fn
         from torch._dynamo.variables.user_defined import InspectVariable
 
         redirected_attrs = []
-        original_var_getattr = InspectVariable.var_getattr
+        original_getattro_impl = InspectVariable.getattro_impl
 
-        def tracking_var_getattr(self, tx, name):
+        def tracking_getattro_impl(self, tx, name):
             redirects = self._PROPERTY_REDIRECTS.get(type(self.value), {})
             if name in redirects:
                 redirected_attrs.append(name)
-            return original_var_getattr(self, tx, name)
+            return original_getattro_impl(self, tx, name)
 
         def fn(x, gn):
             sig = inspect.signature(gn)
@@ -12740,7 +12740,7 @@ fn
             return a + b
 
         x = torch.randn(2, 3)
-        with patch.object(InspectVariable, "var_getattr", tracking_var_getattr):
+        with patch.object(InspectVariable, "getattro_impl", tracking_getattro_impl):
             opt_fn = torch.compile(fn, backend="eager", fullgraph=True)
             result = opt_fn(x, gn)
 
@@ -13124,7 +13124,7 @@ fn
         self.assertEqual(cnts.frame_count, 1)
 
     def test_getattrvariable_as_python_constant(self):
-        from torch._dynamo.variables.misc import GetAttrVariable
+        from torch._dynamo.variables.misc import CallMethodVariable as CMV
 
         @torch.compile(backend="eager")
         def fn(x, rand1):
@@ -13140,7 +13140,7 @@ fn
         x = torch.randn(3, 3)
         expected = fn.__wrapped__(x, get_rng())
 
-        with patch.object(GetAttrVariable, "as_python_constant", autospec=True) as po:
+        with patch.object(CMV, "as_python_constant", autospec=True) as po:
             actual = fn(x, get_rng())
 
         self.assertEqual(expected, actual)

